examples/godot_tps_controller_port/src/player.rs

```diff
@@ -0,0 +1,815 @@
+};
+use gdnative::prelude::*;
+
+/// The player "class"
+#[derive(NativeClass, Debug)]
+#[inherit(KinematicBody)]
+pub struct Player {
+    // CONSTANTS
+    #[property]
+    camera_mouse_rotation_speed: f32,
+
+    #[property]
+    camera_controller_rotation_speed: f32,
+
+    // A minimum angle lower than or equal to -90 breaks movement if the player is looking upward.
+    #[property]
+    camera_x_rot_min: f32,
+
+    #[property]
+    camera_x_rot_max: f32,
+
+    // Release aiming if the mouse/gamepad button was held for longer than 0.4 seconds.
+    // This works well for trackpads and is more accessible by not making long presses a requirement.
+    // If the aiming button was held for less than 0.4 seconds, keep aiming until the aiming button is pressed again.
+    #[property]
+    aim_hold_threshold: f32,
+
+    #[property]
+    direction_interpolate_speed: f32,
+
+    #[property]
+    motion_interpolate_speed: f32,
+
+    #[property]
+    rotation_interpolate_speed: f32,
+
+    #[property]
+    min_airborne_time: f32,
+
+    #[property]
+    jump_speed: f32,
+
+    // VARIABLES
+    #[property]
+    airborne_time: f32,
+
+    #[property]
+    orientation: Transform,
+
+    #[property]
+    root_motion: Transform,
+
+    #[property]
+    motion: Vector2,
+
+    #[property]
+    velocity: Vector3,
+
+    #[property]
+    aiming: bool,
+
+    // If `true`, the aim button was toggled on by a short press (instead of being held down).
+    #[property]
+    toggled_aim: bool,
+
+    // The duration the aiming button was held for (in seconds).
+    #[property]
+    aiming_timer: f32,
+
+    #[property]
+    camera_x_rot: f32,
+
+    // ONREADY VARIABLES
+    #[property]
+    initial_position: Vector3,
+
+    #[property]
+    gravity: Vector3,
+
+    #[property]
+    animation_tree: Option<Ref<AnimationTree>>,
+
+    #[property]
+    player_model: Option<Ref<Spatial>>,
+
+    #[property]
+    shoot_from: Option<Ref<Position3D>>,
+
+    #[property]
+    color_rect: Option<Ref<ColorRect>>,
+
+    #[property]
+    crosshair: Option<Ref<TextureRect>>,
+
+    #[property]
+    fire_cooldown: Option<Ref<Timer>>,
+
+    // Handles Y rotation (yaw)
+    #[property]
+    camera_base: Option<Ref<Spatial>>,
+
+    #[property]
+    camera_animation: Option<Ref<AnimationPlayer>>,
+
+    // Handles x rotation (pitch)
+    #[property]
+    camera_rot: Option<Ref<Spatial>>,
+
+    #[property]
+    camera_spring_arm: Option<Ref<SpringArm>>,
+
+    #[property]
+    camera_camera: Option<Ref<Camera>>,
+
+    #[property]
+    sound_effects: Option<Ref<Node>>,
+
+    #[property]
+    sound_effect_jump: Option<Ref<AudioStreamPlayer>>,
+
+    #[property]
+    sound_effect_land: Option<Ref<AudioStreamPlayer>>,
+
+    #[property]
+    sound_effect_shoot: Option<Ref<AudioStreamPlayer>>,
+
+    #[property]
+    shoot_particle: Option<Ref<Particles>>,
+
+    #[property]
+    muzzle_particle: Option<Ref<Particles>>,
```

Comment:
Is it really necessary that all these be properties? If all of these are always present, consider packing them instead into a struct so you only have to deal with the `Option` once:

```rust
struct SubNodes {
    animation_tree: Ref<AnimationTree>,
    player_model: Ref<Spatial>,
    shoot_from: Ref<Position3D>,
    // snip
}

pub struct Player {
    // snip
    sub_nodes: Option<SubNodes>,
}
```